tests/grid/test_eta.py

```diff
@@ -0,0 +1,216 @@
+is not provided.
+"""
+
+
+def set_answers(eta_file):
+
+    """
+    Read in the expected values of ak and bk
+    arrays from the input eta NetCDF files.
+    """
+
+    data = xr.open_dataset(eta_file)
+    return data["ak"].values, data["bk"].values
+
+
+def write_non_mono_eta_file(in_eta_file, out_eta_file):
+    """
+    Reads in file eta79.nc and alters randomly chosen ak/bk values
+    This tests the expected failure of set_eta_hybrid_coefficients
+    for coefficients that lead to non-monotonically increasing
+    eta values
+    """
+
+    data = xr.open_dataset(in_eta_file)
+    data["ak"].values[10] = data["ak"].values[0]
+    data["bk"].values[20] = 0.0
+
+    data.to_netcdf(out_eta_file)
+
+
+@pytest.mark.parametrize("km", [79, 91])
+def test_set_hybrid_pressure_coefficients_correct(km):
+
+    """This test checks to see that the ak and bk arrays
+    are read-in correctly and are stored as
+    expected.  Both values of km=79 and km=91 are
+    tested and both tests are expected to pass
+    with the stored ak and bk values agreeing with the
+    values read-in directly from the NetCDF file.
+    """
+
+    dirname = os.path.dirname(os.path.abspath(__file__))
+    eta_file = os.path.join(dirname, f"eta{km}.nc")
+
+    backend = "numpy"
+
+    layout = (1,1)
+
+    nz = km
+    ny = 48
+    nx = 48
+    nhalo = 3
+
+    partitioner = CubedSpherePartitioner(TilePartitioner(layout))
+
+    communicator = CubedSphereCommunicator(NullComm(rank=0,total_ranks=6), partitioner)
+
+    sizer = SubtileGridSizer.from_tile_params(
+        nx_tile=nx,
+        ny_tile=ny,
+        nz=nz,
+        n_halo=nhalo,
+        extra_dim_lengths={},
+        layout=layout,
+        tile_partitioner=partitioner.tile,
+        tile_rank=communicator.tile.rank,
+    )
+
+    quantity_factory = QuantityFactory.from_backend(sizer=sizer, backend=backend)
+
+    metric_terms = MetricTerms(quantity_factory=quantity_factory, communicator=communicator, eta_file=eta_file)
+
+    ak_results = metric_terms.ak.data
+    bk_results = metric_terms.bk.data
+    ak_answers, bk_answers = set_answers(f"eta{km}.nc")
+
+    if ak_answers.size != ak_results.size:
+        raise ValueError("Unexpected size of bk")
+    if bk_answers.size != bk_results.size:
+        raise ValueError("Unexpected size of ak")
+
+    if not np.array_equal(ak_answers, ak_results):
+        raise ValueError("Unexpected value of ak")
+    if not np.array_equal(bk_answers, bk_results):
+        raise ValueError("Unexpected value of bk")
+
+
+
+def test_set_hybrid_pressure_coefficients_nofile():
+
+    """This test checks to see that the program
+    fails when (1) the eta_file is not specified in the yaml
+    configuration file; and (2), the computed eta values
+    increase non-monotonically.  For the latter test, the eta_file
+    is specified in test_config_not_mono.yaml file and
+    the ak and bk values in the eta_file have been changed nonsensically
+    to result in erronenous eta values.
```

Comment:
It looks like this test only does the first of these two, so the docstring should reflect that

Reply:
This docstring still needs to be updated